File: models/pharmacy/PrescriptionHeader.js
import mongoose from "mongoose";

const PrescriptionSchema = new mongoose.Schema({

  hospitalId: { type: mongoose.Schema.Types.ObjectId, ref: "Hospital", required: true },
  branchId: { type: mongoose.Schema.Types.ObjectId, ref: "Branch", required: true },

  visitId: { type: mongoose.Schema.Types.ObjectId, ref: "PatientVisit", required: true },
  patientId: { type: mongoose.Schema.Types.ObjectId, ref: "PatientRegistration", required: true },
  doctorId: { type: mongoose.Schema.Types.ObjectId, ref: "User", required: true },

  status: {
    type: String,
    enum: [
      "Pending",
      "SentToPharmacy",
      "PartiallyDispensed",
      "Dispensed",
      "Cancelled"
    ],
    default: "Pending"
  },

  notesForPharmacy: String,
  cancellationReason: String,

  createdBy: { type: mongoose.Schema.Types.ObjectId, ref: "User" }

}, { timestamps: true });

export default mongoose.model("PrescriptionHeader", PrescriptionSchema);

File: jstojv/src/main/java/com/samrat/jstojv/entity/PrescriptionHeader.java
package com.samrat.jstojv.entity;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "prescription_headers")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class PrescriptionHeader extends BaseEntity {

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "hospital_id", nullable = false)
    private Hospital hospital;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "branch_id", nullable = false)
    private Branch branch;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "visit_id", nullable = false)
    private PatientVisit visit;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "patient_id", nullable = false)
    private PatientRegistration patient;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "doctor_id", nullable = false)
    private User doctor;

    @Enumerated(EnumType.STRING)
    private Status status = Status.Pending;

    @Column(name = "notes_for_pharmacy", columnDefinition = "TEXT")
    private String notesForPharmacy;

    @Column(name = "cancellation_reason")
    private String cancellationReason;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "created_by")
    private User createdBy;

    public enum Status {
        Pending, SentToPharmacy, PartiallyDispensed, Dispensed, Cancelled
    }
}
